pkg/collector/corechecks/sbom/check.go

```diff
@@ -200,7 +202,9 @@ func (c *Check) Run() error {
 		case <-containerPeriodicRefreshTicker.C:
 			c.processor.processContainerImagesRefresh(c.workloadmetaStore.ListImages())
 		case <-hostPeriodicRefreshTicker.C:
-			c.processor.processHostRefresh()
+			c.processor.triggerHostScan(hostSbomChan)
+		case scanResult := <-hostSbomChan:
+			c.processor.processHostScanResult(scanResult)
 		case <-metricTicker.C:
 			c.sendUsageMetrics()
 		case <-c.stopCh:
```

Comment:
IMO in this `case`, we can still have a edgecase, when a `c.processor.triggerHostScan()` was call, and delegate the scan to the `processor.sbomScanner` that can try to add later the result in the `hostSbomChan` channel. but in that case the channel will be close. 
So you need to have a protect between `c.processor.triggerHostScan(hostSbomChan)` and `c.processor.processHostScanResult(scanResult)` on solution can be to have maybe a waitGroup between these too `case` but you need to be sure that each time we call `triggerHostScan` the `processHostScanResult` is also call.

Reply:
The channel is never closed. Only the sender has the permission to close the channel. Since this goroutine is only a receiver, what would happen in this case is that the sbom scanner will send the sbom to the channel, that will be buffered then dropped because the receiver goroutine is not listening.

Reply:
Also, the fact that each scan request has an attached channel can be problematic. IMO each collector should have its own channel.